### PR TITLE
Add session template page - fixes

### PR DESCRIPTION
**Pull request by tpmcgowan**
* Fix handling of conditional field for end date
* Set `formValues` in `catch()` block
* Set constant with days of week for re-use across the app

### CODE REVIEW

File: server/views/pages/prisons/addSessionTemplate.njk [hunk 1]
```diff
@@ -24,48 +24,6 @@
   ] %}
 {% set backLinkHref = "/prisons/" + prisonId + "/session-templates" %}
 
-{% set endDateTextHtml %}
```
Comment (tpmcgowan) [inline server/views/pages/prisons/addSessionTemplate.njk:27]: Just moved this close to where it's used so it's easier to see what's going on

File: server/views/pages/prisons/addSessionTemplate.njk [hunk 2]
```diff
@@ -237,10 +203,11 @@
           items: [
             {
               text: "Yes",
+              value: "yes",
```
Comment (tpmcgowan) [inline server/views/pages/prisons/addSessionTemplate.njk:206]: Setting value here makes it easier to see directly if it's been selected

File: server/routes/prisons/sessionTemplates/addSessionTemplateController.ts [hunk 3]
```diff
@@ -49,9 +51,10 @@ export default class AddSessionTemplateController {
         },
         sessionDateRange: {
           validFromDate: `${req.body.validFromDateYear}-${req.body.validFromDateMonth}-${req.body.validFromDateDay}`,
-          validToDate: req.body.hasEndDate
-            ? `${req.body.validToDateYear}-${req.body.validToDateMonth}-${req.body.validToDateDay}`
-            : undefined,
+          validToDate:
+            req.body.hasEndDate === 'yes'
```
Comment (tpmcgowan) [inline server/routes/prisons/sessionTemplates/addSessionTemplateController.ts:55]: Checking here for the value we explicitly set for the conditional checkbox

**Review by hutcheonb-moj — APPROVED** [review]
LGTM

**Pull request merged**